### DISCUSSION TITLE
A large number of buttons overflow

### DISCUSSION THREAD

**Discussion by Zagna** [filamentphp/filament#2383] [Help]
### Package

filament/filament

### Package Version

v2.12.0

### Laravel Version

v9.11.0

### PHP Version

PHP 8.1.5

### Bug description

With enough buttons...
![filament](https://user-images.githubusercontent.com/14203328/167478263-c42d8907-055f-4c81-bcbf-065d743ad856.png)
Without `shrink-0` on the button container div
![filament2](https://user-images.githubusercontent.com/14203328/167478256-4e3dd241-0e40-4310-8dd9-eb9378dd7356.png)
And with some `flex-basis` on the header actions container div
![filament3](https://user-images.githubusercontent.com/14203328/167478266-7e27ceb8-6d27-479f-884d-ec4787e156d2.png)
It manifests with just 6 buttons at certain window sizes

**Comment by ryangjchandler** [comment]
@Zagna We have plans to add a dropdown of some sort if that's potentially helpful?

**Comment by Zagna** [comment]
I have the most jank resource calendar page with the following buttons: Day, Week, Month, Today, Previous and Next.
Desktop size works fine, mobile works fine, overflow happens around portrait tablet.
Maybe for the range select a dropdown could work, just feels more natural as separate buttons.

**Comment by ryangjchandler** [comment]
Even with the shrink class changes, we'd have a hard time making it fit nicely with header text too. @danharrin Thoughts?

**Comment by danharrin** [comment]
It's much more likely to have a long title / heading instead of many actions, so when you test please make sure that you also test out how the actions work with long text like that.

The reason why we have `shrink-0` is so that if the text is too long, the buttons dont shrink.

**Comment by danharrin** [comment]
Hey, I'm gonna move this as I don't think this is an "issue" per-se, but I would accept a PR that adjusts the layout to accommodate for many buttons as well as long headings.

**Comment by Zagna** [comment]
By switching Previous and Next to iconButton, I can save enough space to prevent the overflow in tablet sizes.
My last thought is that maybe heading could switch to flex sooner than sm.
